modules/basegl/src/shadercomponents/volumecomponent.cpp

```diff
@@ -60,10 +64,11 @@ namespace {
 constexpr std::string_view uniforms = util::trim(R"(
 uniform VolumeParameters {0}Parameters;
 uniform sampler3D {0};
+uniform vec4 {0}Zero;
 )");
 
 constexpr std::string_view voxelFirst = util::trim(R"(
-vec4 {0}VoxelPrev = vec4(0);
+vec4 {0}VoxelPrev = vec4({0}Zero);
```

Comment:
Suggest to remove the uniform and use the formula instead
```suggestion
// Initialize previous value with 0 
vec4 {0}VoxelPrev = (0.0 + {0}Parameters.formatOffset) * (1.0 - {0}Parameters.formatScaling);
```

Reply:
How sure are you that that is correct?

Reply:
see https://github.com/inviwo/inviwo/blob/3bbed809efa238fcb1bd29356585e7b295e7d4dd/modules/opengl/glsl/utils/sampler3d.glsl#L43-L54

Reply:
Would probably be good to wrap in a function to make it more clear, voxelDataToNormalized or something

Reply:
But all this does not say if it is the same or not!

Reply:
I know that what I have is correct.... so I don't se the point of spending all this time on a pointless change

Reply:
Yes, they must be the same.

Should it be initialized with zero in the first place? Why not equal to the first sample?

Uniforms can be expensive, at least it was a scarce resource before on GPUs, which could result in slower code. The volume component is a base component so as a precausion I think that it would be better to keep it as slim as possible.
In principle, I don't see a major reason for caching values when they can easily be derived from already uploaded uniforms.

Reply:
consider the other crapton of uniforms we set all the time. I would not be to bothered.

I still don't se why the need to be the same. The formatOffset and formatScaling. take into account the scaling done by opengl it self. Which the uniform I set does not. And should not do unless we change other part of the iso rendering. Hence Im not convinced the are the same.

Reply:
> The formatOffset and formatScaling. take into account the scaling done by opengl it self. Which the uniform I set does not.

I think that they are equivalent. Both should take the "raw" values and transform them to [0 1]. The formatOffset//formatScaling should be adapted to take the internal OpenGL format when doing the transformation. 

> Which the uniform I set does not. And should not do

The iso-surface computation must work in the same value space. Otherwise I don't see how it could be correct. The fact that it does work for you indicate that the computations indeed are the same.